Sources/KnitCodeGen/SourceGen/NamedRegistrationGroup+SourceCode.swift

```diff
@@ -0,0 +1,21 @@
+//
+// Copyright © Block, Inc. All rights reserved.
+//
+
+import Foundation
+import SwiftSyntax
+
+extension NamedRegistrationGroup {
+    // Generate the enum for this group of named registrations
+    func enumSourceCode(assemblyName: String) throws -> DeclSyntaxProtocol {
```

Comment:
I moved this out of TypeSafetySourceFile to make this code easier to test in isolation.